File: javascript/dictionary.js
var fs = require("fs");

var Trie = require("trie-hard");

var dictionary = new Trie();

var words = fs.readFileSync("../words.txt", "utf8");
words.split("\n").map(function(word) {
  return word.trim();
}).forEach(function(word) {
  if (word) {
    dictionary.add(word);
  }
});

module.exports = dictionary;

File: python/dictionary.py
import datrie
import string


class StringTrie(datrie.Trie):

    def __init__(self, *args, **kwargs):
        return super(StringTrie, self).__init__(string.ascii_letters, *args, **kwargs)

    def contains(self, string_):
        """Check if a word exists in the dictionary."""
        return unicode(string_) in self

    def add(self, string_):
        """Add a word to the dictionary."""
        self[unicode(string_)] = ''

dictionary = StringTrie()

with open("../words.txt", "r") as words:
    for word in words.read().splitlines():
        if word:
            dictionary.add(word)
